docs/framework-opentelemetry_javaagent.md

```diff
@@ -0,0 +1,40 @@
+# OpenTelemetry Javaagent
+
+The OpenTelemetry Javaagent buildpack framework will cause an application to be automatically instrumented
+with the [OpenTelemetry Javaagent Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation).
+
+Data will be sent directly to the OpenTelemetry Collector. 
+
+<table>
+  <tr>
+    <td><strong>Detection Criterion</strong></td>
+    <td>Existence of a bound service containing the string <code>otel-collector</code></td>
+  </tr>
+  <tr>
+    <td><strong>Tags</strong></td>
+    <td><code>opentelemetry-javaagent=&lt;version&gt;</code></td>
+  </tr>
+</table>
+
+Tags are printed to standard output by the buildpack detect script
+
+## User-Provided Service
+
+Users are currently expected to provide their own "custom user provided service" (cups) 
```

Comment:
Should I remove `(cups)` too?  It won't match the abbreviation now.

Reply:
Maybe you could just change 'provide' to 'create', because `cups` is the short command for create-user-provided-service 😅

Reply:
Note: this may be less relevant when CF has a collector bundled in the image VM, right?

Reply:
I updated this section a bit.

Reply:
> Note: this may be less relevant when CF has a collector bundled in the image VM, right?

It may be yes, but for the moment the initial built-in solution in CF is not exposed to apps as a collector. Therefore the manual/cups collector is the only option. 

Given this, I think it would be helpful to show an example of a complete user-provided service payload that would allow the app to connect to a collector deployed as an app, e.g. IP/port variables etc.? I don't think the Splunk integration needed this because the agent is configured to connect to a set backend collector service.

Related to this, I noticed that you are not taking any of the properties set in the bound service and applying them as agent/app runtime properties, so for example, in your updated doc:

```
$ cf cups otel-collector -p '{"otel.resource.attributes": "deployment.environment=cf-demo"}'
```

These properties will not be used anywhere as they are not read/applied by the framework. The Splunk integration [does this](https://github.com/cloudfoundry/java-buildpack/blob/bb37d7cdedf0f4c840ba95a21becf96690aa3359/lib/java_buildpack/framework/splunk_otel_java_agent.rb#L38) so you should be able to follow the pattern there.

Reply:
I've applied the config mapping you mentioned, but I'm not sure what an example would look like.

Reply:
Reading the docs as they are currently, it seems a user would need to look up the properties required to connect the agent to a collector, which I think is required for this integration to be useable in this initial offering.

Therefore I was hoping to have a complete example of how to add the agent to an app, which gets config properties from the binding, and sends data to a user-defined collector. I think the last part is all that's left. I can then test the steps in my env, as a user. 

Please let me know if I'm not correct about how this should work!

Reply:
@nrekretep I'm not a cloudfoundry user, so I could use your assistance. I'm not sure how to respond to David's request here.  Do you think you could chime in with a more authoritative response?

Reply:
If understanding correctly then the docs should contain a complete example of the necessary configuration for the user provided `otel-collector` service. 

What needs to be added to the docs depends on which exporters and which protocol you want to use to send data to the collector. 

Lets assume you want to just use the otlp exporter over grpc for metrics and traces and the collector endpoint uses tls and basic auth. 

From my understanding the following properties need to be added to the user provided service:

```json
{
  "otel.exporter.otlp.endpoint" : "https://my-collector-endpoint",
  "otel.exporter.otlp.headers" : "authorization=Basic SOMEBAS64STRING",
  "otel.exporter.otlp.protocol" : "grpc",
  "otel.metrics.exporter" : "otlp",
  "otel.traces.exporter" : "otlp",
}
```

We use this setup in CF, but we configure the otel agent via environment variables. Setting the java system properties as above should work, too. 

HTH

Reply:
This would be ideal @nrekretep - an example of basic/minimum config required for the user-service along with a link to the full options available.

@tylerbenson are you ok to add this to the docs part of your PR? Alternatively I could add this separately.

We're aiming to get this merged and into a release next week.

Reply:
@pivotal-david-osullivan I would really appreciate if you could take this over.  Thanks!